homeassistant/loader.py

```diff
@@ -1153,6 +1153,18 @@ def bind_hass(func: _CallableT) -> _CallableT:
     The use of this decorator is discouraged, and it should not be used
     for new functions.
     """
+    from .helpers.frame import report  # pylint: disable=import-outside-toplevel
+
+    report(
+        (
+            "uses @bind_hass decorator."
+            " This is deprecated and will stop working in Home Assistant 2024.9, it"
+            " should be updated to pass hass object as first argument"
```

Comment:
we should discourage the use of `hass.components` completely. So not just passing `hass` as first argument.